### PR TITLE
GP19-36: Api request

### PR DESCRIPTION
**Pull request by sophiambolah**
made an axios request, implemented loading screen and error handling

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,17 +1,42 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import ConferenceList from "./ConferenceList";
+import { getConferenceList } from "./api";
```
Comment (RaMorey) [inline src/App.js]: This is a very minor comment, but have a think about how you want to structure your imports. It's nice to have an order to them. For example, alphabetical, and then order all of the packages first, then your internal imports, then css

File: src/App.js [hunk 2]
```diff
@@ -1,17 +1,42 @@
...
+    }
+    fetchData();
+  }, []);
+
   return (
     <StyledApp>
-      <ConferenceList conferences={conferences} />
+      {isLoading ? (
+        <p>Loading...</p>
+      ) : (
```
Comment (RaMorey) [inline src/App.js]: this is just a readability thing for me, so don't feel the need to do it, but I quite like organising ternarys like this:

```
isLoading
? <p>Loading...</>
: <ConferenceList conferences={conferences} />
```

File: src/App.js (hunk 1 above)
Comment (MerrettM) [reply src/App.js]: I think I've done this in the latest commit?

**Comment by MerrettM** [comment]
> Does that work without the leading 'http://'?

Ah whoops, no it doesn't!

**Comment by MerrettM** [comment]
Apologies, for that, I clicked the wrong button

**Comment by MerrettM** [comment]
> Does that work without the leading 'http://'?

Ah, no it doesn't, whoops!

File: src/api.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ import axios from "axios";
 
 export const getConferenceList = async () => {
   try {
-    const response = await axios.get("localhost:8080/conferences");
+    const response = await axios.get("http://localhost:8080/conferences");
```
Comment (dex-mendeley) [inline src/api.js]: For now it's probably alright to keep it there. In the future you probably have some config file with those urls :)

**Pull request merged**